### DISCUSSION TITLE
How to get url of each image?

### DISCUSSION THREAD

**Discussion by reinatch** [dzhuang/django-galleryfield#48] [Q&A] [answered]
Hey there, 
I'am getting just an array of ids 

images = [ 0, 2, 3, 5, 6, 7, 7]

I've set up and reciving the images in MEDIA_ROOT, ?

**Comment by dzhuang** [comment, marked as answer]
As documented in [the doc](galleryfield.models.BuiltInGalleryImage), those ids are the `pk`s of the `target_model` instances (which should be `garlleryfield.BuiltInGalleryImage` instances if you did not do model customization). You can query the instances through:

   ```python
    from garlleryfield import BuiltInGalleryImage
    images = [ 0, 2, 3, 5, 6, 7, 7]
    img_urls = []
    for pk in images:
        img_urls.append(BuiltInGalleryImage.objects.get(pk=pk).url)
    print(img_urls)
   ``` 

Or more simpler, if the `Gallery` instance is `my_gallery`, it can be:

   ```python
    img_urls = [img.url for img in my_gallery.images.objects.all()]
   ```

The latter is better as you don't need to know the actual model of the image instances.
